var/spack/repos/builtin/packages/mvapich2-gdr/package.py

```diff
@@ -0,0 +1,189 @@
+    )
+
+    @property
+    def libs(self):
+        query_parameters = self.spec.last_query.extra_parameters
+        libraries = ['libmpi']
+
+        if 'cxx' in query_parameters:
+            libraries = ['libmpicxx'] + libraries
+
+        return find_libraries(
+            libraries, root=self.prefix, shared=True, recursive=True
+        )
+
+    @property
+    def process_manager_options(self):
+        spec = self.spec
+
+        opts = []
+
+        if '~mcast' in spec:
+            opts.append('--disable-mcast')
+
+        if '+core_direct' in spec:
+            opts.append('--with-core-direct')
+
+        if '+openacc' in spec:
+            opts.append('--enable-openacc')
+
+        # See: http://slurm.schedmd.com/mpi_guide.html#mvapich2
+        if 'process_managers=slurm' in spec:
+            opts.append('--with-pm=slurm')
+            if 'pmi_version=pmi1' in spec:
+                opts.append('--with-pmi=pmi1')
+            if 'pmi_version=pmi2' in spec:
+                opts.append('--with-pmi=pmi2')
+            if 'pmi_version=pmix' in spec:
+                opts.append('--with-pmi=pmix')
+                opts.append('--with-pmix={0}'.format(spec['pmix'].prefix))
+
+        elif 'process_managers=pbs' in spec:
+            opts.append([
+                '--with-pm=hydra',
+                '--with-pbs=/opt/pbs'
+            ])
+            if '~mcast' in spec:
+                opts.append('--disable-mcast')
+            if '+core_direct' in spec:
+                opts.append('--with-core-direct')
+            if '+openacc' in spec:
+                opts.append('--enable-openacc')
+
+        elif 'process_managers=jsrun' in spec:
+            opts.append([
+                '--with-pmi=pmix',
+                '--with-pmix={0}'.format(spec['pmix'].prefix),
+                '--with-pm=jsm'
+            ])
+
+        return opts
+
+    def setup_environment(self, spack_env, run_env):
+        spec = self.spec
+        # mvapich2 configure fails when F90 and F90FLAGS are set
+        spack_env.unset('F90')
+        spack_env.unset('F90FLAGS')
+        if 'pmi_version=pmi1' in spec:
+            run_env.set('SLURM_MPI_TYPE', 'pmi1')
+        if 'pmi_version=pmi2' in spec:
+            run_env.set('SLURM_MPI_TYPE', 'pmi2')
+        if 'pmi_version=pmix' in spec:
+            run_env.set('SLURM_MPI_TYPE', 'pmix')
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        spack_env.set('MPICC',  join_path(self.prefix.bin, 'mpicc'))
+        spack_env.set('MPICXX', join_path(self.prefix.bin, 'mpicxx'))
+        spack_env.set('MPIF77', join_path(self.prefix.bin, 'mpif77'))
+        spack_env.set('MPIF90', join_path(self.prefix.bin, 'mpif90'))
+
+        spack_env.set('MPICH_CC', spack_cc)
+        spack_env.set('MPICH_CXX', spack_cxx)
+        spack_env.set('MPICH_F77', spack_f77)
+        spack_env.set('MPICH_F90', spack_fc)
+        spack_env.set('MPICH_FC', spack_fc)
```

Comment:
The unit tests are failing because you're using old APIs (setup_environment, setup_dependent_environment) instead of the new APIs (setup_build_environment, setup_run_environment, setup_dependent_build_environment, etc).

Reply:
I've addressed all flake8 warnings except long lines for version's sha256sum, and URLs.

As for the older APIs, I updated the files according to package.py of MVAPICH2. 

Also, is `from spack import *` required? It seems not to be needed in MVAPICH2 package.py